### PR TITLE
Replace TopBarRegistry with reactive solution

### PR DESCRIPTION
**Pull request by jansav**
This PR will:

1. Remove TopBarRegistry in favor of reactive solution
2. Introduce a way to inject all enabled renderer extensions

### CODE REVIEW

**Comment by jansav** [comment]
Branch also here https://github.com/lensapp/lens/tree/top-bar-registry

File: src/renderer/components/layout/top-bar/top-bar.tsx [hunk 1]
```diff
@@ -48,34 +54,10 @@ ipcRendererOn("history:can-go-forward", (event, state: boolean) => {
   nextEnabled.set(state);
 });
 
-export const TopBar = observer(({ children, ...rest }: Props) => {
+const NonInjectedTopBar: React.FC<Props & Dependencies> = (({ items, children, ...rest }) => {
```
Comment (aleksfront) [inline src/renderer/components/layout/top-bar/top-bar.tsx]: Let's stick with single way of setting `Props` notation as of
```
const NonInjectedTopBar = (({ items, children, ...rest }: Props & Dependencies) => 
```
Unless we have some benefits of using `React.FC`.

**Review by aleksfront — DISMISSED** [review]

File: src/renderer/components/layout/top-bar/top-bar.tsx (hunk 1 above)
Comment (jansav) [reply src/renderer/components/layout/top-bar/top-bar.tsx]: Sure, done.

**Comment by jansav** [comment]
@aleksfront mind approving again.

**Review by Iku-turso — DISMISSED** [review]

**Review by Iku-turso — APPROVED** [review]

**Review by aleksfront — APPROVED** [review]

**Pull request merged**